tox-offline-draft.md

```diff
@@ -5,77 +5,146 @@
 
-(TODO: Inbox location can be changed if necessary, as long as new location
-is always signed by OM key. Is this a useful feature? If you change you
-lose all old messages stored in old locations).
+A user creates a new public and private key pair (Current DHT keys are
+ephemeral) per friend to encrypt offline messages.  Messages are stored
+amongst an array of known nodes.  Users then store messages they want to send
+to a store on those nodes after optionally completing a PoW function. 
 
 ## Retrieving Messages
 
-Users simply check their inbox location in the DHT when they come online.
-If any messages are available they will download them from the nodes
-holding them and decrypt them with their OM private keys. Nodes holding
-messages should them delete them from their store.
-
-(TODO: Similar to what dpush does, we could have the receiver scan sender's
-public keyspaces for messages intended for them, thus requiring no PoW from
-the sender. I feel like this can reintroduce spam though, however the
-receiver is fully in control of what they want to pull at this point.)
+Users simply check the SENDER's STORAGE NODES in the DHT when they come
+online.  If any messages are available they will download them from the nodes
+holding them and decrypt them with their OM keys. Nodes holding messages
+should then delete them from their store.
 
 ## Spam
 
 As a Proof-of-Work (PoW) hash function is used to insert messages into the
-DHT. The sender must complete a certain amount of computational hashing
-before they can send a message. Messages that do not properly complete the
-hashing function will be dropped by network clients upon receival. Thus is
-makes spamming a user unreliable and costly.
-
+DHT. The SENDER and RECEIVER must complete a certain amount of computational
+hashing before they can send a message. Messages that do not properly
+complete the hashing function will be dropped by network clients upon
+receival. Thus is makes spamming a user or the network unreliable and costly.
+
+The difficulty of the PoW is increased relative to the storage space required
+to storage messages, to ensure an attacker cannot simply fill a
+STORAGE_NODE's available message space with a single message. However, very
+small messages could game this multiplier, and thus even the minimum PoW
+difficulty must be reasonably large.
+
+Messages should also expire eventually, and thus a higher PoW difficulty
+should be required for a longer message TTL (time to live).
+        
 ## Privacy
 
 Tox does not inherently provide anonymity, however this will not expose any
 more information about the user than using the DHT in Tox already does.
-(TODO: A correlation between who is talking to whom may be possible?).
+
+A middle node can correlate which IPs are talking to each other by seeing who
+requests messages sent by whom, however one could enumerate this simply by
+watching traffic from either the SENDER or RECEIVER so this does not provide
+much new information.
```

Comment:
I'd be interested to hear about that new protocol. Also yes new onion paths or old nodes should be fine here.